webapp/src/components/markdownEditorInput/markdownEditorInput.tsx

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {
+    ReactElement,
+    useEffect,
+    useMemo,
+    useCallback,
+    useRef,
+    useState,
+} from 'react'
+import {getDefaultKeyBinding, EditorState, ContentState, DraftHandleValue} from 'draft-js'
+import Editor from '@draft-js-plugins/editor'
+import createMentionPlugin, {
+    defaultSuggestionsFilter,
+    MentionData,
+} from '@draft-js-plugins/mention'
+import '@draft-js-plugins/mention/lib/plugin.css'
+import './markdownEditorInput.scss'
+
+import createEmojiPlugin from '@draft-js-plugins/emoji'
+import '@draft-js-plugins/emoji/lib/plugin.css'
+
+import {getWorkspaceUsersList} from '../../store/users'
+import {useAppSelector} from '../../store/hooks'
+import {IUser} from '../../user'
+
+const imageURLForUser = (window as any).Components?.imageURLForUser
+
+type Props = {
+    onChange?: (text: string) => void
+    onFocus?: () => void
+    onBlur?: (text: string) => void
+    initialText?: string
+    id?: string
+    isEditing: boolean
+}
+
+const MarkdownEditorInput = (props: Props): ReactElement => {
+    const {onChange, onFocus, onBlur, initialText, id, isEditing} = props
+    const workspaceUsers = useAppSelector<IUser[]>(getWorkspaceUsersList)
+    const mentions: MentionData[] = useMemo(() => workspaceUsers.map((user) => ({name: user.username, avatar: `${imageURLForUser ? imageURLForUser(user.id) : ''}`})), [workspaceUsers])
+    const ref = useRef<Editor>(null)
+    const [editorState, setEditorState] = useState(() => {
+        const state = EditorState.createWithContent(ContentState.createFromText(initialText || ''))
+        return EditorState.moveSelectionToEnd(state)
+    })
+    const [isMentionPopoverOpen, setIsMentionPopoverOpen] = useState(false)
+    const [isEmojiPopoverOpen, setIsEmojiPopoverOpen] = useState(false)
+    const [suggestions, setSuggestions] = useState(mentions)
+
+    const {MentionSuggestions, plugins, EmojiSuggestions} = useMemo(() => {
+        const mentionPlugin = createMentionPlugin({mentionPrefix: '@'})
+        const emojiPlugin = createEmojiPlugin()
+
+        // eslint-disable-next-line no-shadow
+        const {EmojiSuggestions} = emojiPlugin
+        // eslint-disable-next-line no-shadow
+        const {MentionSuggestions} = mentionPlugin
+        // eslint-disable-next-line no-shadow
+        const plugins = [
+            mentionPlugin,
+            emojiPlugin,
+        ]
+        return {plugins, MentionSuggestions, EmojiSuggestions}
+    }, [])
+
+    useEffect(() => {
+        setTimeout(() => ref.current?.focus(), 200)
+    })
```

Comment:
@hahmadia why do we need this `useEffect`? It breaks setting focus to card title textarea.

Reply:
@kamre Good call. I'll fix this in a PR.
We need it so that when you are in editing mode, it auto focuses on the DraftJS editor. Otherwise, it won't if this is not called.

Reply:
@hahmadia I have created pull request #1952 with the proposed fix, please review it.